src/Model/Profile.php

```diff
@@ -642,9 +642,21 @@ public static function getEvents()
 		$classtoday = '';
 
 		$s = dba::p(
-			"SELECT `event`.* FROM `event`
-			WHERE `event`.`uid` = ? AND `type` != 'birthday' AND `start` < ? AND `start` >= ?
-			ORDER BY `start` ASC ",
+			"SELECT `event`.*, `item`.`id`, `item`.`thr-parent`, `item`.`parent-uri`
+			FROM `event`
+			LEFT JOIN `item`
```

Comment:
This query seems to be a little complicated to me. Why not doing it that way:
```SELECT * FROM `event` WHERE `type` != 'birthday' AND NOT EXISTS(SELECT `id` FROM `item` WHERE `item`.`uid` = `event`.`uid` AND `item`.`parent-uri` = `event`.`uri` AND verb='http://purl.org/zot/activity/attendno');```

Reply:
When you cancel your attendance status, the item stays, so you need the check on `visible` and `deleted`.

I can check the performance of NOT EXISTS but I’m afraid it wouldn’t scale well with the number of potential events. I made sure the LEFT JOIN was very fast because the query is executed on each network page load.

Reply:
In my tests it looked great. Possibly you can check via "explain" as well?

Reply:
Nevermind, I ran both query without time constraints, mine ran in 6 seconds, yours in 0.3 seconds.

Reply:
Just a little difference ;-)